planner/dispatcher.go

```diff
@@ -0,0 +1,51 @@
+package planner
+
+import (
+	"context"
+	"github.com/weihesdlegend/Vacation-planner/iowrappers"
+	"sync"
+)
+
+const NumWorkers = 10
+
+type Dispatcher struct {
+	JobQueue chan *iowrappers.Job
+	workers  []*PlanningSolutionsWorker
+	solver   *Solver
+	wg       *sync.WaitGroup
+}
+
+func NewDispatcher(s *Solver) *Dispatcher {
+	return &Dispatcher{
+		JobQueue: make(chan *iowrappers.Job),
+		workers:  make([]*PlanningSolutionsWorker, 0),
+		solver:   s,
+		wg:       &sync.WaitGroup{},
+	}
+}
+
+func (d *Dispatcher) Run(ctx context.Context) {
+	mu := &sync.RWMutex{}
```

Comment:
This is nice, so basically regardless of how many users create new Jobs, they are always locked !